OpenRA.Mods.Common/FileFormats/WavReader.cs

```diff
@@ -66,6 +66,9 @@ public static bool LoadSound(Stream s, out Func<Stream> result, out short channe
 						blockAlign = s.ReadInt16();
 						sampleBits = s.ReadInt16();
 
+						if (audioType == WaveType.MsAdpcm)
+							sampleBits = 16; // Unsure why this is different to the value above, but it needs to be 16 (!)
```

Comment:
Is this a general MSADPCM thing, or a specific issue with the remaster files?

I would really like to understand what is going on here before we merge this, because I suspect there should be a more general and elegant fix!

Reply:
From https://icculus.org/SDL_sound/downloads/external_documentation/wavecomp.htm's section on ` WAVE_FORMAT_ADPCM`:

> **wBitsPerSample** This is the number of bits per sample of ADPCM. Currently only 4 bits per sample is defined. Other values are reserved.

So this is indeed supposed to not be 16 (i confirmed that the remaster files are indeed using 4), and we must be using this value incorrectly in the code below.

Reply:
Below the switch case we have a similar override `if (audioType == WaveType.ImaAdpcm)`.

I'm going to guess that the rest of the sound code is using `sampleBits` differently to how the sound format defines, so the best way forward here is probably to delete this override and change the one below to `audioType != WaveType.PCM`.

Reply:
> (but would be nice if someone could point at a spec definition confirming it)

From http://www.cs.columbia.edu/~hgs/audio/dvi/IMA_ADPCM.pdf:

> The  DVI®  ADPCM  algorithm  was selected  to  provide  a  standard  method  to  exchange  high  quality  (16-bit  signed linear)  data  at  data  rates  one-half  that  of  the  uncompressed  8-bit  data  rates  at  a given sampling frequency

This confirms that ADPCM will always have 4 bits per input sample expanding to 16 bits per output sample.